challenge.rb

```diff
@@ -1,26 +1,26 @@
 
 def capitalize_each_string(input)
-  #implement your solution here
+input.map { |n| n.capitalize }
 end
 
 def fetch_the_dog(input)
-  #implement your solution here
+  input.select { |n| n.downcase == "dog" }
 end
 
 def no_dogs_allowed(input)
-  #implement your solution here
+input.reject{ |n| n.downcase == "dog" }
 end
 
 def count_the_animals(input)
-  #implement your solution here
+  input.count
 end
 
 def fetch_the_first_two(input)
-  #implement your solution here
+  input.first(2)
 end
 
 def fetch_CD_animals(input)
-  #implement your solution here
+  input.find_all { |n| n.downcase == "dog" || n.downcase == "cat" }
```

Comment:
cool. you can also use `.grep` which takes a regex pattern. it's super handy.
https://ruby-doc.org/core-2.5.0/Enumerable.html#method-i-grep